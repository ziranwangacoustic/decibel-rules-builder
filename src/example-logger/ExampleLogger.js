/* **************************************************************
 *
 * Copyright (C) 2020 Acoustic, L.P. All rights reserved.
 *
 * NOTICE: This file contains material that is confidential and proprietary to
 * Acoustic, L.P. and/or other developers. No license is granted under any intellectual or
 * industrial property rights of Acoustic, L.P. except as may be provided in an agreement with
 * Acoustic, L.P. Any unauthorized copying or distribution of content from this file is
 * prohibited.
 *
 *************************************************************** */

import { createLoggerFactory } from '@cui/common-services';
import loggerConfig from '../loggerConfig';

export const exampleLoggerCfg = loggerConfig;

const { LoggerProvider, useLogger } = createLoggerFactory();

export { LoggerProvider, useLogger };
