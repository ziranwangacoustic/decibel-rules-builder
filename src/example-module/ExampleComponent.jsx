/* **************************************************************
 *
 * Copyright (C) 2020 Acoustic, L.P. All rights reserved.
 *
 * NOTICE: This file contains material that is confidential and proprietary to
 * Acoustic, L.P. and/or other developers. No license is granted under any intellectual or
 * industrial property rights of Acoustic, L.P. except as may be provided in an agreement with
 * Acoustic, L.P. Any unauthorized copying or distribution of content from this file is
 * prohibited.
 *
 *************************************************************** */

import React from 'react';
import { Button, FullPageTemplate, HR, H2, H3, UnorderedList, ListItem } from '@decibel/components';
import { useDispatch, useSelector } from 'react-redux';
import { useStatusMessages, useDispatchRequest } from '@cui/shell-ui';
import { actions } from './reducers';
import { useLogger } from '../example-logger/ExampleLogger';
import useTranslation from '../i18n';
import { EXAMPLE_URL } from './epics';

export default () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const log = useLogger('ExampleComponent');
  const { showSuccessMessage } = useStatusMessages();

  const { fetch } = useDispatchRequest(
    {
      url: EXAMPLE_URL,
      successType: actions.successRequest.type,
    },
    { trigger: false },
  );
  const onPingClickHandler = () => {
    log.info('ping');
    showSuccessMessage({ messageContent: 'PING message' });
    dispatch(actions.ping());
  };

  const onRequestClickHandler = () => {
    fetch();
  };

  const isPinging = !!useSelector(state => state.pingReducer.isPinging);

  return (
    <FullPageTemplate header="Home Page">
      <H2>PING / PONG example</H2>
      <H3>{`is pinging: ${isPinging.toString()}`}</H3>
      <Button type="button" onClick={onPingClickHandler}>
        Start PING
      </Button>
      <Button type="button" onClick={onRequestClickHandler}>
        Just make a request
      </Button>
      <HR />
      <H2>Translation example</H2>
      <UnorderedList nested>
        <ListItem>{t('Hello {{name}}', { name: 'DynamicUserName' })}</ListItem>
        <ListItem>{t('Sample text')}</ListItem>
        <ListItem>{t('Not translated text (missing translation)')}</ListItem>
      </UnorderedList>
    </FullPageTemplate>
  );
};
