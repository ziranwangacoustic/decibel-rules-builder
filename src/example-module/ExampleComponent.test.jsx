/* **************************************************************
 *
 * Copyright (C) 2020 Acoustic, L.P. All rights reserved.
 *
 * NOTICE: This file contains material that is confidential and proprietary to
 * Acoustic, L.P. and/or other developers. No license is granted under any intellectual or
 * industrial property rights of Acoustic, L.P. except as may be provided in an agreement with
 * Acoustic, L.P. Any unauthorized copying or distribution of content from this file is
 * prohibited.
 *
 *************************************************************** */

/* eslint-disable no-unused-vars */
import axios from 'axios';
import React, { Suspense } from 'react';
import { Provider } from 'react-redux';
import { createEpicMiddleware } from 'redux-observable';
import { fireEvent, render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router';
import { LoggerProvider } from '../example-logger/ExampleLogger';
import ExampleComponent from './ExampleComponent';
import reducer, { actions } from './reducers';

const middlewares = [createEpicMiddleware()];
const mockStore = configureMockStore(middlewares);

jest.mock('axios');

const createWrapper = storeInstance => ({ children }) => (
  <Suspense fallback="loading">
    <Provider store={storeInstance}>
      <MemoryRouter>
        <LoggerProvider>{children}</LoggerProvider>
      </MemoryRouter>
    </Provider>
  </Suspense>
);

describe('ExampleComponent', () => {
  let storeInstance;
  beforeEach(() => {
    storeInstance = mockStore(reducer);
  });

  test('should match snapshots', async () => {
    const { container, findByText } = render(<ExampleComponent />, {
      wrapper: createWrapper(storeInstance),
    });
    await findByText('Start PING');
    expect(container.firstChild).toMatchSnapshot();
  });

  test('ping pong integration', async () => {
    const { getByText, findByText } = render(<ExampleComponent />, {
      wrapper: createWrapper(storeInstance),
    });

    const pingButton = await findByText('Start PING');
    expect(getByText('is pinging: false')).toBeInTheDocument();

    fireEvent(
      pingButton,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    expect(storeInstance.getActions()).toMatchObject([
      expect.anything(),
      { type: actions.ping.type, payload: undefined },
    ]);
  });
});
