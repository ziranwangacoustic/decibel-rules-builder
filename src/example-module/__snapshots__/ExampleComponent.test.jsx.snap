// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent should match snapshots 1`] = `
<main
  class="bx--full-page-template"
  style=""
>
  <div
    class="bx--full-page-template__heading"
  >
    <div
      class="bx--full-page-template__header-left"
    >
      <div
        class="bx--full-page-template__header-wrapper"
      >
        <h2
          class=" bx--heading_default_color bx--heading_2"
        >
          Home Page
        </h2>
      </div>
    </div>
  </div>
  <div
    class="bx--full-page-template__content"
  >
    <h2
      class=" bx--heading_default_color bx--heading_2"
    >
      PING / PONG example
    </h2>
    <h3
      class=" bx--heading_default_color bx--heading_3"
    >
      is pinging: false
    </h3>
    <button
      class="bx--btn bx--btn--primary"
      tabindex="0"
      type="button"
    >
      Start PING
    </button>
    <button
      class="bx--btn bx--btn--primary"
      tabindex="0"
      type="button"
    >
      Just make a request
    </button>
    <div
      aria-hidden="true"
      class=" bx--hr"
    />
    <h2
      class=" bx--heading_default_color bx--heading_2"
    >
      Translation example
    </h2>
    <ul
      class="bx--list--unordered bx--list--nested"
    >
      <li
        class="bx--list__item"
      >
        Hello, DynamicUserName (in English)
      </li>
      <li
        class="bx--list__item"
      >
        This is sample text (in English)
      </li>
      <li
        class="bx--list__item"
      >
        Not translated text (missing translation)
      </li>
    </ul>
  </div>
</main>
`;
